### ISSUE TITLE
Question about NPM performance

### ISSUE THREAD

**Issue by SongHoSung** [facebookresearch/NPM#2]
Hi~ . Thank you for introducing an interesting research.
I have a few questions about npm's performance.
  
1. In the performance of NPM, is there a result of distinguishing the effect of only aligning the context of the batch or the effect of configuring the contrastive learning? There would be a benefit from learning similar documents together, but I am curious about the effect of excluding it.

2. Is there any result of learning npm from scratch without post-training in roberta? I think it would be difficult to learn NPM without understanding the basic language.

3. About npm fine tuning (you said it was a future work),
  It showed amazing performance in zero-shot, but fine tuning is not expected to improve the performance of the part that retrieves similar syntax to the reference corpus or the part that matches the mask token based on the retrieved reference. 
  What do you think?
  
 Thanks~!

**Comment by shmsw25** [comment]
Thanks for your interest to our work and for kind words! (And sorry about the late reply -- I was on vacation and just got back.)

Below is my answer to your questions:

1. If I understand correctly, do you mean if we have ablations on batching, e.g., grouping same documents so that they share similar context vs. random batching? We tried it in the very early version of the paper but did not try with the final version of NPM. We'll add this ablation in the next version of the paper -- thanks for suggesting!

2. We are planning to explore training NPM without initialization from RoBERTA. Based on our early experiments, performance isn't as good as NPM from RoBERTa or RoBERTa, but is better than the RoBERTa baseline that we trained ourselves. I think this is because the original RoBERTa is trained on massive batch size with 512 GPUs (we used 32 GPUs), so it's hard to beat the original RoBERTa.

3. I think it depends on how you fine-tune the model. For instance, you can train the model to be a retrieval model (as work in dense retrieval does like [DPR](https://arxiv.org/abs/2004.04906) and [DensePhrases](https://arxiv.org/abs/2012.12624)), and in this case the model will be learned to be better at retrieving the right tokens/phrases from the reference corpus.

**Comment by SongHoSung** [comment]
Thank you for your detailed answer.

**Comment by SongHoSung** [comment]
Thank you for your detailed answer.
